modules/platform/nuxeo-apidoc-server/nuxeo-apidoc-webengine/src/main/java/org/nuxeo/apidoc/browse/Distribution.java

```diff
@@ -0,0 +1,593 @@
+        String pkgList = formData.getString("packages");
+        SnapshotFilter filter = new SnapshotFilter(distribLabel);
+
+        if (bundleList != null) {
+            String[] bundles = bundleList.split("\n");
+            for (String bundleId : bundles) {
+                filter.addBundlePrefix(bundleId);
+            }
+        }
+
+        if (pkgList != null) {
+            String[] packages = pkgList.split("\\r?\\n");
+            for (String pkg : packages) {
+                filter.addPackagesPrefix(pkg);
+            }
+        }
+
+        Map<String, Serializable> otherProperties = readFormData(formData);
+
+        log.info("Start Snapshot...");
+        boolean startedTx = false;
+        UserTransaction tx = TransactionHelper.lookupUserTransaction();
+        if (tx != null && !TransactionHelper.isTransactionActiveOrMarkedRollback()) {
+            tx.begin();
+            startedTx = true;
+        }
+        try {
+            getSnapshotManager().persistRuntimeSnapshot(getContext().getCoreSession(), distribLabel, otherProperties,
+                    filter);
+        } catch (NuxeoException e) {
+            log.error("Error during storage", e);
+            if (tx != null) {
+                tx.rollback();
+            }
+            return getView("savedKO").arg("message", e.getMessage());
+        }
+        log.info("Snapshot saved.");
+        if (tx != null && startedTx) {
+            tx.commit();
+        }
+        return getView("saved");
+    }
+
+    /**
+     * Returns the runtime snapshot json export.
+     *
+     * @since 11.1
+     */
+    @GET
+    @Path("json")
+    @Produces("application/json")
+    public Object getJson() throws IOException {
+        // init potential resources depending on request
+        getSnapshotManager().initWebContext(getContext().getRequest());
+        DistributionSnapshot snap = getSnapshotManager().getRuntimeSnapshot();
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        snap.writeJson(out);
+        return out.toString();
+    }
+
+    public String getDocumentationInfo() {
+        DocumentationService ds = Framework.getService(DocumentationService.class);
+        return ds.getDocumentationStats(getContext().getCoreSession());
+    }
+
+    protected File getExportTmpFile() {
+        File tmpFile = new File(Environment.getDefault().getTemp(), "export.zip");
+        if (tmpFile.exists()) {
+            tmpFile.delete();
+        }
+        tmpFile.deleteOnExit();
+        return tmpFile;
+    }
+
+    @GET
+    @Path("downloadDoc")
+    public Response downloadDoc() throws IOException {
+        DocumentationService ds = Framework.getService(DocumentationService.class);
+        File tmp = getExportTmpFile();
+        tmp.createNewFile();
+        OutputStream out = new FileOutputStream(tmp);
+        ds.exportDocumentation(getContext().getCoreSession(), out);
+        out.flush();
+        out.close();
```

Comment:
You should use try-with-resources:
```java
try (OutputStream out = new FileOutputStream(tmp)) {
    ds.exportDocumentation(getContext().getCoreSession(), out);
    out.flush();
}
```

Reply:
indeed but i only moved that code -> i would prefer not modifying it to avoid unforeseen regressions in these logics